mozdownload/scraper.py

```diff
@@ -244,6 +244,62 @@ def detect_platform(self):
         else:
             return "%s%d" % (mozinfo.os, mozinfo.bits)
 
+    @property
+    def checksums_binary(self):
+        """Return the name of the checksums file."""
+        def _get_binary():
+            # Retrieve all entries from the remote virtual folder
+            parser = self._create_directory_parser(self.checksums_path)
+            if not parser.entries:
+                raise errors.NotFoundError('No entries found', self.checksums_path)
+
+            # Download the first matched directory entry
+            pattern = re.compile(self.checksums_binary_regex, re.IGNORECASE)
+            for entry in parser.entries:
+                try:
+                    self._checksums_binary = pattern.match(entry).group()
+                    break
+                except Exception:
+                    # No match, continue with next entry
+                    continue
+            else:
+                raise errors.NotFoundError("Checksums binary not found in folder",
+                                           self.checksums_path)
+
+        self._retry_check_404(_get_binary)
+
+        return self._checksums_binary
+
+    @property
+    def checksums_binary_regex(self):
+        """Return the regex for the checksums filename."""
+        raise errors.NotImplementedError(sys._getframe(0).f_code.co_name)
+
+    @property
+    def checksums_url(self):
+        """Return the URL of the checksums."""
+        return urllib.quote(urljoin(self.checksums_path, self.checksums_binary),
+                            safe='%/:=&?~#+!$,;\'@()*[]|')
+
+    @property
+    def checksums_path(self):
+        """Return the path to the checksums folder."""
+        return urljoin(self.base_url, self.checksums_path_regex)
+
+    @property
+    def checksums_path_regex(self):
+        """Return the regex for the path to the checksums folder."""
+        raise errors.NotImplementedError(sys._getframe(0).f_code.co_name)
+
+    @property
+    def checksums_parse_regex(self):
+        return r"(.+)\s.+\s(.+)\s(.+)"
+
+    @property
+    def checksums_binary_name(self):
```

Comment:
The checksum file contains a list of all the files in the current remote folder. So I would propose that we have a method which accepts a filename and then returns all the checksum data for that file. To not clutter the scraper class with all those details a new checksum class should be created. All the properties and methods having the `checksum` prefix indicate that already.